evap/results/tests/test_tools.py

```diff
@@ -190,3 +192,74 @@ def test_result_calculation_with_no_contributor_rating_question_does_not_fail(se
+        cls.general_contribution = cls.course.general_contribution
+        cls.general_contribution.questionnaires.set([cls.questionnaire])
+        cls.responsible1_contribution = mommy.make(Contribution, contributor=cls.responsible1, course=cls.course,
+            questionnaires=[cls.questionnaire], responsible=True, can_edit=True, comment_visibility=Contribution.ALL_COMMENTS)
+        cls.responsible2_contribution = mommy.make(Contribution, contributor=cls.responsible2, course=cls.course,
+            questionnaires=[cls.questionnaire], responsible=True, can_edit=True, comment_visibility=Contribution.ALL_COMMENTS)
+        cls.contributor_own_contribution = mommy.make(Contribution, contributor=cls.contributor_own, course=cls.course,
+            questionnaires=[cls.questionnaire], comment_visibility=Contribution.OWN_COMMENTS)
+        cls.contributor_course_contribution = mommy.make(Contribution, contributor=cls.contributor_course, course=cls.course,
+            questionnaires=[cls.questionnaire], comment_visibility=Contribution.GENERAL_COMMENTS)
+        cls.contributor_all_contribution = mommy.make(Contribution, contributor=cls.contributor_all, course=cls.course,
+            questionnaires=[cls.questionnaire], comment_visibility=Contribution.ALL_COMMENTS)
+        cls.general_contribution_textanswer = mommy.make(TextAnswer, question=cls.question, contribution=cls.general_contribution, state=TextAnswer.PUBLISHED)
+        cls.responsible1_textanswer = mommy.make(TextAnswer, question=cls.question, contribution=cls.responsible1_contribution, state=TextAnswer.PUBLISHED)
+        cls.responsible2_textanswer = mommy.make(TextAnswer, question=cls.question, contribution=cls.responsible2_contribution, state=TextAnswer.PUBLISHED)
+        cls.contributor_own_textanswer = mommy.make(TextAnswer, question=cls.question, contribution=cls.contributor_own_contribution, state=TextAnswer.PUBLISHED)
+        cls.contributor_course_textanswer = mommy.make(TextAnswer, question=cls.question, contribution=cls.contributor_course_contribution, state=TextAnswer.PUBLISHED)
+        cls.contributor_all_textanswer = mommy.make(TextAnswer, question=cls.question, contribution=cls.contributor_all_contribution, state=TextAnswer.PUBLISHED)
+
+    def test_textanswer_visibility_info(self):
+        contributions = [
+            self.general_contribution, self.responsible1_contribution, self.responsible2_contribution,
+            self.contributor_own_contribution, self.contributor_course_contribution, self.contributor_all_contribution
+        ]
+        textanswers = [
+            self.general_contribution_textanswer, self.responsible1_textanswer, self.responsible2_textanswer,
+            self.contributor_own_textanswer, self.contributor_course_textanswer, self.contributor_all_textanswer
+        ]
+        visible_to = [textanswers_visible_to(contribution) for contribution in contributions]
```

Comment:
you can remove the `contributions` list, use `len(textanswers)` everywhere, and access textanswer.contribution here.